### PR TITLE
[Servnig] Fix plot when before param exists.

### PR DESCRIPTION
**Pull request by davesh0812**
[ML-3835](https://jira.iguazeng.com/browse/ML-3835)

### CODE REVIEW

File: mlrun/serving/states.py [hunk 1]
```diff
@@ -1291,6 +1291,15 @@ def _add_graphviz_flow(
                 else {}
             )
             graph.edge(previous_object.fullname, child.fullname, **kw)
+        before = (child.before if hasattr(child, "before") else []) or []
```
Comment (gtopper) [inline mlrun/serving/states.py:1294]:
```suggestion
        before = getattr(child, "before", [])
```

Comment (davesh0812) [reply mlrun/serving/states.py]: thanks

File: mlrun/serving/states.py [hunk 2]
```diff
@@ -1291,6 +1291,15 @@ def _add_graphviz_flow(
...
             )
             graph.edge(previous_object.fullname, child.fullname, **kw)
+        before = getattr(child, "before", [])
+        for item in before:
+            next_object = step[item]
+            kw = (
+                {"ltail": "cluster_" + next_object.fullname}
+                if next_object.kind == StepKinds.router
+                else {}
+            )
```
Comment (gtopper) [inline mlrun/serving/states.py:1301]: Small thing, but I think this is a bit clearer, and makes it easier to add keywords in the future
```suggestion
            kw = {}
            if next_object.kind == StepKinds.router:
                kw["ltail"] = f"cluster_{next_object.fullname}"
```

**Review by gtopper — COMMENTED** [review]
It seems to make sense, but I think we need a test, or at least a reproduce procedure (self contained snippet or notebook). And a before and after example.

This will be requested anyway when QA get the ticket.

File: mlrun/serving/states.py (hunk 2 above)
Comment (davesh0812) [reply mlrun/serving/states.py]: fixed and uploaded a notebook

**Review by gtopper — APPROVED** [review]
Looks good to me. Maybe in the future we can get some tests for `graph.plot()`.

**Pull request merged**